File: iOS-Native/MentalHealthCompanion/Services/ChatService.swift
import Foundation
import Combine
// MARK: - Chat Service (matching chat_service.dart)
class ChatService: ObservableObject {

    private let baseURL = "https://gen-ai-g6tt.onrender.com/api/v1/chat"

    func sendMessages(userId: String, message: String) async -> String {
        guard let url = URL(string: "\(baseURL)/add") else { return "" }

        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")

        let body: [String: Any] = ["userId": userId, "message": message]
        request.httpBody = try? JSONSerialization.data(withJSONObject: body)

        do {
            let (data, response) = try await URLSession.shared.data(for: request)
            guard let httpResponse = response as? HTTPURLResponse,
                  httpResponse.statusCode == 200 else {
                print("Failed to send message")
                return ""
            }
            print("Message sent")
            if let json = try? JSONSerialization.jsonObject(with: data) as? [String: Any],
               let botResponse = json["botResponse"] as? String {
                return botResponse
            }
            return ""
        } catch {
            print("Error sending messages: \(error)")
            return ""
        }
    }

    func getMessages(userId: String, date: String) async -> [[String: Any]] {
        guard let url = URL(string: "\(baseURL)/get/\(userId)/\(date)") else { return [] }

        do {
            let (data, response) = try await URLSession.shared.data(from: url)
            guard let httpResponse = response as? HTTPURLResponse,
                  httpResponse.statusCode == 200 else {
                print("Failed to get messages")
                return []
            }
            if let messages = try? JSONSerialization.jsonObject(with: data) as? [[String: Any]] {
                return messages
            }
            return []
        } catch {
            print("Error getting messages: \(error)")
            return []
        }
    }
}

File: app/lib/services/chat_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:saathi/services/api_helpers.dart';

class ChatService {
  Future<String> sendMessages(String userId, String message) async {
    try {
      final url = Uri.parse('https://gen-ai-g6tt.onrender.com/api/v1/chat/add');
      final headers = {'Content-Type': 'application/json'};
      final body = jsonEncode({
        'userId': userId,
        'message': message,
      });
      final response = await http.post(url, headers: headers, body: body);
      if (response.statusCode == 200) {
        final responseBody = safeJsonDecode(response);
        return responseBody['botResponse'] ?? '';
      } else {
        throw BackendException('Failed to send message', statusCode: response.statusCode);
      }
    } catch (e) {
      print("Error sending messages: $e");
      rethrow;
    }
  }

  Future<List<dynamic>> getMessages(String userId, String date) async {
    return retryWithBackoff(() async {
      final url = Uri.parse(
          'https://gen-ai-g6tt.onrender.com/api/v1/chat/get/$userId/$date');
      final response = await http.get(url);
      if (response.statusCode == 200) {
        final messages = safeJsonDecode(response);
        if (messages is List) {
          return messages;
        } else {
          throw BackendException('Invalid message list format', statusCode: response.statusCode);
        }
      } else {
        throw BackendException('Failed to get messages', statusCode: response.statusCode);
      }
    });
  }
}
